Sources/WebSocketKit/WebSocket.swift

```diff
@@ -272,22 +327,65 @@ public final class WebSocket: Sendable {
             } else {
                 self.close(code: .protocolError, promise: nil)
             }
+            
         case .text, .binary:
-            // create a new frame sequence or use existing
-            self.frameSequence.withLockedValue { currentFrameSequence in
-                var frameSequence = currentFrameSequence ?? .init(type: frame.opcode)
-                // append this frame and update the sequence
-                frameSequence.append(frame)
-                currentFrameSequence = frameSequence
+            // is compressed? and have pmce configured for the socket?
+            if frame.rsv1 , let pmce = pmce, pmce.enabled {
+               
+                do {
+                  
+                    if frame.maskKey != nil {
+                        
+                        let newFrame = try pmce.unmaskedDecompressedUnamsked(frame: frame)
+                        // create a new frame sequence or use existing
+                        self.frameSequence.withLockedValue { currentFrameSequence in
+                            var frameSequence = currentFrameSequence ?? .init(type: frame.opcode)
+                            // append this frame and update the sequence
+                            frameSequence.append(newFrame)
+                            currentFrameSequence = frameSequence
+                        }
+                    }
+                    else {
+                      
+                        let newFrame = try pmce.decompressed(frame)
+                        // create a new frame sequence or use existing
+                        self.frameSequence.withLockedValue { currentFrameSequence in
+                            var frameSequence = currentFrameSequence ?? .init(type: frame.opcode)
+                            // append this frame and update the sequence
+                            frameSequence.append(newFrame)
+                            currentFrameSequence = frameSequence
+                        }
+                    }
+                }
+                catch {
+                    print("websocket-kit: \(error)")
+                }
             }
+            else if frame.rsv1 && pmce == nil {
+                if pmce?.logging ?? false {
+                    print("websocket-kit: PMCE:  received compressed frame without PMCE configured! Closing per RFC-7692. You could have a configuration issue.")
```

Comment:
Use swift-log

Reply:
That would mean adding a dep to websocket kit for it I assume. I had looked into it possibly but thought it kinda out of scope and thought if ppl wanted logger there they would have one by now and maybe there was a reason websocket had no logger.

Reply:
How is logging handled right now in websocket-kit? Simply no logging or with prints? If it's no logging I would keep it that way and make a separate PR eventually adding proper logging and the dependency.

Reply:
It was no logging. I asked Tim and he agreed it shold be added. I can't make another fork though so I just did it on this PR. It was kind of a pain point to debug it without logging so I added what I could after asking around for a while. But logging has been added via swift log to websocket and pmce classes with a boxed bool to enable with a defeat of false. that way you can turn it on for real time issues without needed a fork to add it.

Reply:
Logging is added now on main.